101-vm-with-data-management-gateway/azuredeploy.json

```diff
@@ -0,0 +1,297 @@
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the virtual machine"
+      }
+    },
+    "existingVirtualNetworkName":{
+      "type": "string",
+      "metadata": {
+        "description": "your existing vnet name"
+      }
+    },
+    "existingVnetLocation":
+    {
+      "type": "string",
+      "allowedValues": [
+        "Central US",
+        "East US",
+        "East US 2",
+        "North Central US",
+        "South Central US",
+        "West Central US",
+        "West US",
+        "West US 2",
+        "North Europe",
+        "West Europe",
+        "East Asia",
+        "Southeast Asia",
+        "Japan East",
+        "Japan West",
+        "Brazil South",
+        "Australia East",
+        "Australia Southeast",
+        "Central India",
+        "South India",
+        "Canada India",
+        "Canada East",
+        "Germany Central",
+        "Germany Northeast",
+        "UK South",
+        "UK West"
+      ],
+      "metadata": {
+        "description": "your virtual machine will be create in the same datacenter with VNET"
+      }
+    },
+    "existingVnetResourceGroupName":{
+      "type": "string",
+      "metadata": {
+        "description": "Name of the existing VNET resource group"
+      }
+    },
+    "existingSubnetInYourVnet": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the subnet in the virtual network you want to use"
+      }
+    },
+    "enableToSetDataStorePasswordsFromInternet": {
+      "type": "string",
+      "allowedValues": ["yes", "no"],
+      "defaultValue": "yes",
+      "metadata": {
+        "description": "If you choose yes, we will create a dns name label for your machine and open one inbound port for our service to access your machine and set data store passwords."
+      }
+    },
+    "_artifactsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "The base URI where artifacts required by this template are located."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/xiaoyingLJ/azure-quickstart-templates/master/101-vm-with-data-management-gateway/"
+    }
+  },
+  "variables": {
+    "baseUri": "[parameters('_artifactsLocation')]",
+    "nsglink": "[concat(variables('baseUri'), 'nested/', parameters('enableToSetDataStorePasswordsFromInternet'),'IncomingRemote.json')]",
+    "nsgTemplateName": "[concat(parameters('virtualMachineName'), 'NSGTemplate')]",
+    "prefix": "[split(parameters('virtualMachineName'), '-')[0]]",
+    "storageAccountName": "[take(concat(toLower(variables('prefix')), uniquestring(resourceGroup().id, parameters('virtualMachineName'))), 24)]",
+    "nicName": "[concat(parameters('virtualMachineName'), 'Nic')]",
+    "publicIPAddressName": "[concat(parameters('virtualMachineName'), '-ip')]",
+    "networkSecurityGroupName": "[concat(parameters('virtualMachineName'), 'nsg')]",
```

Comment:
For both PIP and NSG, this is a very fragile pattern - these values should be passed through to the nested deployments rather than requiring variables to be defined the same way in different files